lib/palindrome_permutation.rb

```diff
@@ -1,5 +1,7 @@
 
 def palindrome_permutation?(string)
-
- 
+  return true if string.empty?
+  max = string.length.even? ? string.length/2 : (string.length/2) + 1
+  palindrome_check = Hash[*string.split('').collect{ |a| [a, 0] }.flatten]
+  return palindrome_check.length == max
```

Comment:
This does not account for the case where there are more than 2 instances of a letter in the palindrome.  In the case of "aaaa", the hash will only contain {"a" => 0} and will have length 1, but max will be set to 2.

Reply:
Whoops, that can be quickly fixed by changing the return: 

`  return palindrome_check.length == max || palindrome_check.length == 1`

Reply:
Sorry, my comments earlier were unclear, this is not actually an edge case issue, rather the issue is a problem with the underlying algorithm.  For example, if the string is "aaaabc" the hash generated by line 5 will be {"a"=>0, "b"=>0, "c"=>0}, and max will equal 3 because the string is of length 6, so this algorithm will return true when it should return false.  Conversely, if the string is "aaaabbbb" the hash generated will be {"a"=>0, "b"=>0}, max will be 4, length of hash will be 2 and the algorithm will return false when it should return true.  To successfully check if a string can be rearranged into a palindrome, either every character must show up an even number of times or every character except one must show up an even number of times.  If you want to discuss this further, I'd be happy to talk to you on slack or we can schedule a video chat if that would be helpful.

Reply:
Ahh, makes sense. Let me see what I can come up with :))